src/ui/toast.js

```diff
@@ -53,121 +40,99 @@ export const toastImportantStyles = {
  */
 export let ToastSpecDef;
 
+/** @const {!Object<string, string>} */
+const iframeAttributes = {
+  'frameborder': '0',
+  'scrolling': 'no',
+  'class': 'swg-toast',
+};
 
 /**
  * The class Notification toast.
  */
 export class Toast {
 
   /**
-   * @param {!Window} win
-   * @param {!ToastSpecDef} spec
+   * @param {!../runtime/deps.DepsDef} deps
+   * @param {{'source': string}} args
    */
-  constructor(win, spec) {
+  constructor(deps, args) {
 
     /** @private @const {!Window} */
-    this.win_ = win;
+    this.win_ = deps.win();
 
     /** @private @const {!HTMLDocument} */
-    this.doc_ = win.document;
+    this.doc_ = this.win_.document;
+
+    /** @private @const {!web-activities/activity-ports.ActivityPorts} */
+    this.activityPorts_ = deps.activities();
 
-    /** @private @const {!ToastSpecDef} */
-    this.spec_ = spec;
+    /** @private @const {!../runtime/deps.DepsDef} */
+    this.deps_ = deps;
 
-    /** @private @const {!FriendlyIframe} */
-    this.iframe_ = new FriendlyIframe(this.doc_, {'class': 'swg-toast'});
+    /** @private @const {{'source': string}} */
+    this.args_ = args;
 
-    /** @private {?Element} */
-    this.container_ = null;
+    /** @private @const {!HTMLIFrameElement} */
+    this.iframe_ =
+        /** @type {!HTMLIFrameElement} */ (
+            createElement(this.doc_, 'iframe', iframeAttributes));
 
-    setImportantStyles(this.iframe_.getElement(), toastImportantStyles);
-    setStyles(this.iframe_.getElement(), topFriendlyIframePositionStyles);
+    setImportantStyles(this.iframe_, toastImportantStyles);
+    setStyles(this.iframe_, topFriendlyIframePositionStyles);
+
+            /** @private @const {!Promise} */
+    this.ready_ = new Promise(resolve => {
+      this.iframe_.onload = resolve;
+    });
   }
 
-  /**
-   * Gets the attached iframe instance.
-   * @return {!FriendlyIframe}
-   */
-  getIframe() {
-    return this.iframe_;
+  getElement() {
+    return this.iframe;
   }
 
   /**
-   * Gets the Iframe element.
-   * @return {!HTMLIFrameElement}
+   * When promise is resolved.
+   * @return {!Promise}
    */
-  getElement() {
-    return this.iframe_.getElement();
+  whenReady() {
```

Comment:
Do we have to expose this method? It seems a bit ambiguous given that there are several notions of readiness. I think the promise returned by `open()` should be enough?

Reply:
Removed whenReady() function call.